### PR TITLE
Dframework-ui latest changes

### PR DESCRIPTION
**Pull request by Collins87mbathi**

### CODE REVIEW

File: src/lib/components/Grid/CustomDropdownmenu.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import { useState } from 'react';
+import {
+    useGridSelector,
+    gridFilterModelSelector,
```
Comment (Pramodspraxa) [inline src/lib/components/Grid/CustomDropdownmenu.js]: Remove extra comma.

File: src/lib/components/Grid/CustomDropdownmenu.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    const apiRef = props.apiRef;
+    const filterModel = useGridSelector(apiRef, gridFilterModelSelector);
+
+    const currentFieldFilters = React.useMemo(
+        () =>
+            filterModel.items?.filter((value) => {
+                if (value.operator === 'isAnyOf') {
+                    setIsMultiSelect(true)
```
Comment (Pramodspraxa) [inline src/lib/components/Grid/CustomDropdownmenu.js]: semicolon missing.

File: src/lib/components/Grid/LocalizedDatePicker.js [hunk 3]
```diff
@@ -3,38 +3,84 @@ import dayjs from 'dayjs';
...
+    const filterFormat = fixedFilterFormat[columnType];
+    const isValidDate = (date) => {
+        const parsedDate = dayjs(date);
+        return parsedDate.isValid() && parsedDate.year() > 1900;
+    };
+    const format = (systemDateTimeFormat((columnType !== "datetime" && columnType !== "dateTimeLocal", false, stateData.dateTime)));
+
+    const handleFilterChange = (newValue) => {
+        if (columnType !== "date" && columnType !== "datetime" && columnType !== "dateTimeLocal") return;
+        const isPartialDate = (value) => {
```
Comment (Pramodspraxa) [inline src/lib/components/Grid/LocalizedDatePicker.js:32]: why function inside a function ?

File: src/lib/components/useRouter/StateProvider.js [hunk 4]
```diff
@@ -30,6 +34,42 @@ const StateProvider = ({ children }) => {
     }
     return isDateFormatOnly ? 'DD-MM-YYYY' : 'DD-MM-YYYY hh:mm:ss A';
   }
+  async function getAllSavedPreferences({ preferenceName, Username, history, dispatchData, preferenceApi }) {
+    let params = {
```
Comment (Pramodspraxa) [inline src/lib/components/useRouter/StateProvider.js]: params should be const

File: src/lib/components/useRouter/StateProvider.js [hunk 5]
```diff
@@ -30,6 +34,42 @@ const StateProvider = ({ children }) => {
...
+      action: 'list',
+      id: preferenceName,
+      Username
+    }
+    const response = await request({ url: preferenceApi, params, history, dispatchData });
+    dispatchData({ type: actionsStateProvider.UDPATE_PREFERENCES, payload: response?.preferences });
+    dispatchData({ type: actionsStateProvider.TOTAL_PREFERENCES, payload: response.totalCount });
+  }
+  async function applyDefaultPreferenceIfExists({ gridRef, history, dispatchData, Username, preferenceName, setIsGridPreferenceFetched, preferenceApi }) {
+    let params = {
```
Comment (Pramodspraxa) [inline src/lib/components/useRouter/StateProvider.js]: it should be const

File: src/lib/components/useRouter/stateReducer.js [hunk 6]
```diff
@@ -19,9 +19,15 @@ const stateReducer = (state, action) => {
...
-            case actionsStateProvider.ACOSTA_REPORT_COLUMNS:
-                return { ...state, getAcostaColumns: action.payload };
-        default:
+        case actionsStateProvider.UDPATE_PREFERENCES:
+            return { ...state, preferences: action.payload }
+        case actionsStateProvider.SET_CURRENT_PREFERENCE_NAME:
+            return { ...state, currentPreference: action.payload }
+        case actionsStateProvider.TOTAL_PREFERENCES:
+            return { ...state, totalPreferences: action.payload }
+                case actionsStateProvider.UPDATE_LOADER_STATE:
```
Comment (Pramodspraxa) [inline src/lib/components/useRouter/stateReducer.js]: formatting missing.

File: src/lib/components/utils.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    filterFieldDataTypes: {
+        Number: 'number',
+        String: 'string',
+        Boolean: 'boolean'
+    },
+    fixedFilterFormat: {
+        date: "YYYY-MM-DD",
+        datetime: "YYYY-MM-DD hh:mm:ss a",
+        dateTimeLocal: "YYYY-MM-DD hh:mm:ss a",
+        OverrideDateFormat: "DD-MMM-YYYY",
```
Comment (Pramodspraxa) [inline src/lib/components/utils.js]: Remove extra comma

**Review by Pramodspraxa — CHANGES_REQUESTED** [review]

File: src/lib/components/Grid/LocalizedDatePicker.js (hunk 3 above)
Comment (Collins87mbathi) [reply src/lib/components/Grid/LocalizedDatePicker.js:32]: thats how I found it written , I think its a conditional function

**Review by aasanntiwari — APPROVED** [review]

**Pull request merged**